src/main/resources/org/cbioportal/legacy/persistence/mybatis/MolecularDataMapper.xml

```diff
@@ -30,24 +30,50 @@
         </where>
     </select>
 
+    <sql id="geneticAlterationSubQueryWithProjection">
+        SELECT
+            genetic_alteration.genetic_entity_id as "genetic_entity_id",
+            genetic_alteration.genetic_profile_id as "genetic_profile_id",
+            gp.stable_id as "stable_id"
+            <if test="projection == 'DETAILED'">
+                ,
+                <include refid="org.cbioportal.legacy.persistence.mybatis.GeneMapper.select">
+                    <property name="prefix" value="GENE."/>
+                </include>
+            </if>
+        FROM genetic_alteration ga
+        INNER JOIN genetic_profile gp
+            ON genetic_alteration.genetic_profile_id = genetic_profile.genetic_profile_id
+        INNER JOIN gene g
+            ON ga.genetic_entity_id = g.genetic_entity_id    
+    </sql>
+    
+    <sql id="geneticAlterationSubQuery">
+        SELECT
+            genetic_alteration.genetic_entity_id as "genetic_entity_id",
+            genetic_alteration.genetic_profile_id as "genetic_profile_id",
+            gp.stable_id as "stable_id"
+        FROM genetic_alteration ga
+        INNER JOIN genetic_profile gp
+            ON genetic_alteration.genetic_profile_id = genetic_profile.genetic_profile_id
+        INNER JOIN gene g
+            ON ga.genetic_entity_id = g.genetic_entity_id
+    </sql>
+    
     <!-- Any changes to this routine should be kept in sync with getGeneMolecularAlterationsIter below -->
     <select id="getGeneMolecularAlterations" resultType="org.cbioportal.legacy.model.GeneMolecularAlteration">
-        SELECT
-        gene.entrez_gene_id AS "entrezGeneId",
-        genetic_alteration.`values` AS "values"
-        <if test="projection == 'DETAILED'">
-            ,
-            <include refid="org.cbioportal.legacy.persistence.mybatis.GeneMapper.select">
-                <property name="prefix" value="GENE."/>
-            </include>
-        </if>
-        FROM genetic_alteration
-        INNER JOIN genetic_profile ON genetic_alteration.genetic_profile_id = genetic_profile.genetic_profile_id
-        INNER JOIN gene ON genetic_alteration.genetic_entity_id = gene.genetic_entity_id
+        WITH dd AS (<include refid="geneticAlterationSubQueryWithProjection" />)
+        SELECT 
+            ga.values,
```

Comment:
@onursumer maybe the fact that values is first (instead of entrezGeneId) affects ordering or response and could account for difference?

Reply:
This was happening because we were incorrectly using genetic entity id instead of entrez gene id. This has been fixed by modifying the SQL.